### DISCUSSION TITLE
Auto snooze if you add treatment

### DISCUSSION THREAD

**Discussion by spanosp** [NightscoutFoundation/xDrip#3134] [Ideas]
It could be useful to have the option for automatic snoozing if you log a treatment

**Comment by Navid200** [comment]
Are you saying if you eat something and enter the corresponding carbs in xDrip, then, xDrip would avoid alerting you even if a low alert is triggered later?

That is not a safe functionality and could lead to a disaster.

**Reply by spanosp** [reply]
Not exaclty, I'm saying that the alarm will be snoozed. Pressumably when you enter a new treatment (carbs or insulin) you are already aware of your BG levels.

**Reply by Navid200** [reply]
This could open the door to lots of possible unintended consequences.  

You can enter a treatment for someone you follow.  There are tech savvy people who do that for their partners.
After they enter the treatment, the diabetic partner may still need the alert.  If entering the treatment snoozes the alert, that may cause problems for the individual.

If you are entering a treatment, it means you are already using your phone, swiping a notification right to snooze it does not take a significant time.

I cannot feel the need for this.

**Reply by spanosp** [reply]
Had not considered the implications in the case of master/follower etc

This option would be useful when the alarm has not triggered yet but will do so very soon afterwards.

**Reply by spanosp** [reply]
Hi, allow me to get back to this after some thought

Its very often when I see my BG rising, I give myself some insulin and my BG continues to rise and triggers an alert shortly afterwards.
When the alert triggers I am still waiting for the insulin to kick-in and will wait before I give myself more so I just snooze the alert.

This is why I think it would be useful if the alarm snoozed (maybe a snooze shorter than default) when you input a treatment.

Could this be a feature when a treatment is entered by the master not the follower?
or an option when you input the treatment?

TY

**Reply by Navid200** [reply]
This is a common experience almost everyone who is not on low-carb diet experiences.
You may be looking at the solution at an uncommon place.

We have a persistent high alert.  It only triggers if your readings are higher than your HIGH value more than a set amount of time. 
You can use it to know when you are high when you didn't expect to be high.

Manipulating the snooze is not a good idea because it could interfere with those who use the app not your way.

**Reply by spanosp** [reply]
Persistent High Alerts would help in the scenario described above only if High Alerts are disabled, otherwise High Alerts will trigger anyway ..and disabling High Alerts is not a great idea.

I understand manipulating snooze can be tricky in certain scenarios but maybe it can be an option / master only?

It seams logical to me that once you are aware of your BG getting out of range and you input a treatment to counter this, the alarm can be delayed somewhat.

TY

**Reply by Navid200** [reply]
A high alert is meant to notify you that your reading has crossed a threshold.
I would like to get that notification whether I expect it or not.
What I do after that notification is my choice.  Snoozing an alert takes a few seconds.
You may be looking at an alert as something that requires you to run and do something.  Not everyone does.  
Sometimes, I know that I am going to get a high alert.  But, I am not sure exactly if I will get it in 15 minutes or half an hour.  I want to hear it.  Wouldn't you?

The persistent high alert does not have a dedicated threshold.  It uses the High value.  xDrip also uses that same high value for its statistical analysis.  I don't believe this is the best setup.  I have discussed this with the developers and told them that I would like to add a dedicated threshold to the persistent high alert to remove the tie between statistics and persistent high alert.
I have been told to go ahead.
This is about months ago.  I still have not got to implement that.  
My only point in this paragraph is that there is a lot to do but, there are only a few of us.

We also need to keep in mind what changes are very simple and what changes require very complicated conditionals.
What you are asking for is simple in functionality.  But, implementing it in code will complicate xDrip and create serious dependencies between different sections.  
How complicated something is going to make xDrip is a factor we cannot overlook when deciding if we should go ahead or not.

We need to prioritize taking many different factors into consideration.  

Taking all of that into account, may I ask why you cannot just snooze the alert when it triggers?
